src/fetch/webfinger.rs

```diff
@@ -117,6 +117,75 @@ pub fn build_webfinger_response(subject: String, url: Url) -> Webfinger {
     }
 }
 
+/// Builds a webfinger response similar to `build_webfinger_response` that can have multiple
+/// actors and an optional type.
+///
+/// `urls` takes a vector of tuples. The first item of the tuple is the URL while the second
+/// item is the type, such as `"Person"` or `"Group"`. If `None` is passed for the type, the field
+/// will be empty.
+///
+/// ```
+/// # use url::Url;
+/// # use activitypub_federation::fetch::webfinger::build_webfinger_response_with_type;
+/// let subject = "acct:nutomic@lemmy.ml".to_string();
+/// let url = Url::parse("https://lemmy.ml/u/nutomic")?;
+/// build_webfinger_response_with_type(subject, vec![(url, None)]);
+/// # Ok::<(), anyhow::Error>(())
+/// ```
+///
+/// ```
+/// # use url::Url;
+/// # use activitypub_federation::fetch::webfinger::build_webfinger_response_with_type;
+/// let subject = "acct:nutomic@lemmy.ml".to_string();
+/// let url = Url::parse("https://lemmy.ml/u/nutomic")?;
+/// build_webfinger_response_with_type(subject, vec![(url, Some("Person"))]);
+/// # Ok::<(), anyhow::Error>(())
+/// ```
+/// ```
+/// # use url::Url;
+/// # use activitypub_federation::fetch::webfinger::build_webfinger_response_with_type;
+/// let subject = "acct:asklemmy@lemmy.ml".to_string();
+/// let url = Url::parse("https://lemmy.ml/c/asklemmy")?;
+/// build_webfinger_response_with_type(subject, vec![(url, Some("Group"))]);
+/// # Ok::<(), anyhow::Error>(())
```

Comment:
I would make it clearer in the text that this is meant for cases where multiple types of actors share the same namespace, and names can collide (because webfinger only has a single namespace for everything).

Also it would be better to give a single example which passes urls for both group and person in a single call.

Reply:
I've updated the docs to be clearer on the usage.